Mathlib/LinearAlgebra/Matrix/SVD/Defs.lean

```diff
@@ -0,0 +1,479 @@
+/-
+Copyright (c) 2023 Mohanad Ahmed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed
+-/
+
+import Mathlib.Data.Matrix.ColumnRowPartitioned
+import Mathlib.Data.Matrix.Rank
+import Mathlib.LinearAlgebra.Matrix.SVD.Reindex
+
+/-! # Singular Value Decomposition
+
+This file provides the SVD theorem which decomposes a real/complex matrix into left
+eigenvectors, singular values block diagonal matrix and right eigenvectors.
+
+Any matrix A (M × N) with rank r = A.rank and with elements in the field ℝ or ℂ can be decomposed
+into three matrices:
+  U: an M × M matrix containing the left eigenvectors of the matrix
+  S: an M × N matrix with an r × r block in the upper left corner with nonzero singular values
+  V: an N × N matrix containing the right eigenvectors of the matrix
```

Comment:
This will be formatted as markdown, so should be at least
```suggestion
  * U: an M × M matrix containing the left eigenvectors of the matrix
  * S: an M × N matrix with an r × r block in the upper left corner with nonzero singular values
  * V: an N × N matrix containing the right eigenvectors of the matrix
```
to prevent it all going on one line!

I'd also recommend you either write `M × N matrix` as lean code (`` `Matrix M N 𝕂` ``) or as LaTeX (`$M × N$ matrix`). Same for all the mathematical expressions in the comments.